### PR TITLE
Add validation for registration details on user service

### PR DESCRIPTION
**Pull request by McNaBry**
## Description

- Add validation for registration details on user service using zod
- Add bad request for invalid username/email/password

## Checklist

- [x] I have updated documentation
- [x] All tests passing

## Screenshots (if applicable)

### CODE REVIEW

File: services/user/src/types/custom.ts [hunk 1]
```diff
@@ -1,8 +1,38 @@
...
+    password: z
+        .string({
+            invalid_type_error: UserValidationErrors.INVALID,
+            required_error: UserValidationErrors.REQUIRED,
+        })
+        .regex(
+            /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.{8,})(?=.*[!"#$%&'()*+,-.:;<=>?@\\/\\[\]^_`{|}~])/,
+            UserValidationErrors.INVALID,
+        ),
+});
```
Comment (samuelim01) [inline services/user/src/types/custom.ts:38]: Should we use multiple regexes to express multiple errors?

Comment (McNaBry) [reply services/user/src/types/custom.ts:38]: I was assuming this was just for simple validation and we wouldn't be differentiating between the different errors. Is there a use case you have in mind or is this for readability and maintainability?

Comment (samuelim01) [reply services/user/src/types/custom.ts:38]: Ok yea I should not review PRs at 2am I think this is fine

**Review by samuelim01 — APPROVED** [review]
LGTM

**Pull request merged**